pkg/tbtc/node.go

```diff
@@ -345,3 +347,158 @@ func (n *node) joinDKGIfEligible(seed *big.Int, startBlockNumber uint64) {
 		logger.Infof("not eligible for DKG with seed [0x%x]", seed)
 	}
 }
+
+// joinSigningIfEligible takes a message and undergoes the process of tECDSA
+// signing if this node's operator proves to control some signers of the
+// requested wallet. This is an interactive process, and joinSigningIfEligible
+// can block for an extended period of time while it completes the operation.
+//
+// TODO: Ultimately, one client will handle multiple signature requests for
+//       multiple wallets. Because of that, logging within that function
+//       must use the context of the given signature request.
+func (n *node) joinSigningIfEligible(
+	message *big.Int,
+	walletPublicKey *ecdsa.PublicKey,
+	startBlockNumber uint64,
+) {
+	logger.Infof(
+		"checking eligibility for signature of message [%v]",
+		message,
+	)
+
+	if signers := n.walletRegistry.getSigners(
+		walletPublicKey,
+	); len(signers) > 0 {
+		logger.Infof(
+			"joining signature of message [%v] and "+
+				"controlling [%v] signers",
+			message,
+			len(signers),
+		)
+
+		// All signers belong to one wallet. Take that wallet from the
+		// first signer.
+		wallet := signers[0].wallet
+		// Actual wallet signing group size may be different from the
+		// `GroupSize` parameter of the chain config.
+		signingGroupSize := len(wallet.signingGroupOperators)
+		// The dishonest threshold for the wallet signing group must be
+		// also calculated using the actual wallet signing group size.
+		signingGroupDishonestThreshold := signingGroupSize -
+			n.chain.GetConfig().HonestThreshold
```

Comment:
This piece of code feels like something that should be extracted out of `node.go` into some more specialized component.

Reply:
I will leave that for now and think about it at the end, once retries and other auxiliary signing components are implemented.